Add reload bullets function

maitanaka opened the pull request:
# What
- Edit GunController.cs
  - set reloading method only if the bullets number is under 30
  - play reloading sound

- Edit BulletController.cs
  - manage the number of bullets and bullet boxes
  - set method called in GunController.cs when reload ... count the bullets and bullet boxes

# Why
To add reload function

maitanaka committed:
Edit GunController to play sound of reloading bullets when the number of bullets is less than 30 & press “R”;

maitanaka committed:
Edit BulletContoller.cs to manage the number of bullets;
- define bulletCount and bulletBoxCount in public
- set ReloadBullets method to count bullets and bullet boxes
  - called in GunController.cs when reload

maitanaka committed:
adjust the gameobjects’ position

pchan52 commented on Assets/Scripts/GunController.cs:
```diff
@@ -4,21 +4,23 @@
-	private GameObject bulletFire;
+	private AudioSource reloadAudioSource;
 	private float bulletInterval;
 	private BulletController bulletController;
```
public宣言して playerをアタッチするとGetComponentする必要ないね！

maitanaka replied:
何に対するgetComponentについてでしょうか？><

pchan52 commented on Assets/Scripts/GunController.cs:
```diff
@@ -4,21 +4,23 @@
 
 	private Vector3 bulletHitPosition;
-	private AudioSource audioSource;
+	private AudioSource fireAudioSource;
```
銃をうったときの音もgunからだしていいよ！
AudioClipで出す音を分けよう！

pchan52 commented on Assets/Scripts/GunController.cs:
```diff
@@ -33,14 +35,21 @@ public class GunController : MonoBehaviour {
+				bulletController.ReloadBullets ();
+				reloadAudioSource.PlayOneShot (reloadAudioClip);
+			}
+		}
```
Reload 関数化しよう！

maitanaka committed:
GunController to play shooting sound by switching the audioClip
Transfer reload method from BulletController to GunController
Remove audioSource from bulletFire

pchan52 commented on Assets/Scripts/GunController.cs:
```diff
@@ -33,14 +32,26 @@ public class GunController : MonoBehaviour {
+		if(Input.GetKey(KeyCode.R)){
+			if (bulletController.bulletCount < 30) {
+				Reload ();
+				AudioSource.PlayOneShot (reloadAudioClip);
```
リロード音もメソッドの中でいいかな！

pchan52 commented on Assets/Scripts/GunController.cs:
```diff
@@ -4,21 +4,20 @@
 
-	[SerializeField] GameObject Bulletfire;
+	[SerializeField] GameObject bulletFire;
 	[SerializeField] GameObject player;
```
ここの型を BulletControllerにしてplayerをアタッチしたら、19行目でGetComponentしなくていいはず！
みたところ他のとこでplayerをつかってないので，GameObject型じゃなくてもよさそうだから！

maitanaka committed:
Publicly defined player in bulletController and included sound playing in Reload method.

maitanaka committed:
adjust the scene and objects

pchan52 commented:
良いと思う！
printでbulletCountとbulletBoxCountをだして動画でみせてもらえる？

maitanaka commented:
右手のbulletCountとBulletBoxCountで数がみれます！

https://gyazo.com/699ba75dd7f624ce1dc9d2532362ba21

pchan52 commented:
![oyabaka_lgtm](https://cloud.githubusercontent.com/assets/19141898/23449172/806484ee-fe98-11e6-9ff5-e1df28b7bd1b.jpeg)
